.code-samples.meilisearch.yaml

```diff
@@ -47,15 +47,27 @@ delete_documents_1: |-
   }));
 search_post_1: |-
   client.index("movies").search("American ninja");
-get_update_1: |-
-  client.index("movies").getUpdate(1).getStatus();
-get_all_updates_1: |-
-  for (UpdateStatus update: client.index("movies").getUpdates())
-  {
-    update.getStatus();
-  }
-get_keys_1: |-
-  //Not yet implemented
+get_task_by_index_1: |-
+  client.index("movies").getTask(1);
+get_all_tasks_1: |-
+  client.getTasks();
+get_task_1: |-
+  client.getTask(1);
+get_all_tasks_by_index_1: |-
+  client.index("movies").getTasks();
+get_one_key_1: |-
+  client.getKey("d0552b41536279a0ad88bd595327b96f01176a60c2243e906c52ac02375f9bc4");
+get_all_keys_1: |-
+  client.getKeys();
+create_a_key_1: |-
+  client.createKey(
```

Comment:
Heeey I know that is probably too late and is my fault, but I'm not sure if this is the best way to expose this API to the client.

I was expecting something like this: https://github.com/meilisearch/meilisearch-dotnet/blob/main/.code-samples.meilisearch.yaml#L229

I agree that is way easier to create a function that accepts a String than another that accepts a complex object, in this case, a `Key`.

I invite you to think more about the user trying to create a key using our SDK:

```java
// in a random java program:
public class Main {
  static void myMethod() {
    Date today = new Date();
    Date tomorrow = new Date(today.getTime() + (1000 * 60 * 60 * 24));

    Client client = new Client(new Config("http://127.0.0.1:7700", "masterKey"));

    Map<Integer, String> colours = new HashMap<>();
    colours.put("description", "blue");
    colours.put("indexes", ["colors"]);
    colours.put("actions", ["documents.add"]);
    colours.put("expiresAt", tomorrow);
        
    Gson gson = new Gson();
        
    String options = gson.toJson(colours);

    client.createKey(options);
  }
}
```

Now think what happens if this user doesn't have automated testing to ensure the behavior and it tries to upgrade our SDK to a breaking change version.
This breaking change rename the "indexes" attribute to "mainIndexes" (it's a bad idea but you just for exemplify)

It should not have any error preventing them to put this upgrade in production because we're not ensuring the structure of the types they should use.

But what happens we did something like this:

```java
// the same random java program:
public class Main {
  static void myMethod() {
    Date today = new Date();
    Date tomorrow = new Date(today.getTime() + (1000 * 60 * 60 * 24));

    Client client = new Client(new Config("http://127.0.0.1:7700", "masterKey"));

    Key key = new Key();
    key.setDescription("blue");
    key.setIndexes(["colors"]);
    key.setAction(["documents.add"]);
    key.setExpiresAt(tomorrow);

    client.createKey(key); // of course inside this we will need to call gson and parse ourselves the body
  }
}
```

The `javac` will warn the user about the change because probably it should be using the `setAction` anymore or even so, as we don’t expose our internal API to the client, we could handle it internally in the `gson` parsing we will do inside the `createKey`.
And in this case that `breaking-change` is no longer a `breaking-change`.

Reply:
You are so right I'm going to change that!